src/databricks/labs/remorph/reconcile/connectors/secrets.py

```diff
@@ -0,0 +1,23 @@
+import logging
+
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.errors import NotFound
+from databricks.sdk.service.workspace import GetSecretResponse
+
+logger = logging.getLogger(__name__)
+
+
+class SecretsMixin:
+    _ws: WorkspaceClient
+    _secret_scope: str
+
+    def _get_secret_if_exists(self, secret_key: str) -> GetSecretResponse | None:
+        """Get the secret value given a secret scope & secret key. Log a warning if secret does not exist"""
+        try:
+            # Return the decoded secret value in string format
+            secret = self._ws.secrets.get_secret(self._secret_scope, secret_key)
```

Comment:
It gives the secret response, not the actual secret string.

Reply:
Updated the function.

Reply:
@vijaypavann-db

Reply:
not sure why, but I can't see these changes

Reply:
I think we should use `self._ws.dbutils.secrets.get` which returns direct value rather than using `ws.get_secret` and `base64.b64decode`

Reply:
@vijaypavann-db  take this refactor as part of your Generate Recon PR.